### ISSUE TITLE
[Go] Is github.com/antihax/optional necessary? 

### ISSUE THREAD

**Issue by adamdecaf** [OpenAPITools/openapi-generator#1176]
##### Description

Hey! Thanks for the OpenAPI v3 support with Go. I was wondering about [`github.com/antihax/optional`](https://github.com/antihax/optional) after generating a Go client. Is that library necessary? Could it just be unexported methods?

I ask because that project has one commit with no readme. What happens if that repository is taken over and a nefarious commit is pushed? 

I'll offer to port that code over into the generate itself. 

##### openapi-generator version

v3.3.0

##### OpenAPI declaration file content or url

https://github.com/moov-io/api

**Comment by wing328** [comment]
cc @antihax

**Comment by antihax** [comment]
Best option is to vendor or fork it and modify the template.

The problem is we can't predict the package name inside of the template.

**Comment by adamdecaf** [comment]
What about shipping the functions needed in a file (write it with the template)? i.e. `optional.go`

**Comment by antihax** [comment]
> What about shipping the functions needed in a file (write it with the template)? i.e. optional.go
The path to the package is still unknown and I avoided putting them in the same package because it could (and did) collide with an API specification.

I think it would be better to put this into the OpenAPITools repo so it at least is known to come from the generator.

**Comment by wing328** [comment]
> I think it would be better to put this into the OpenAPITools repo so it at least is known to come from the generator.

Sure we can fork `github.com/antihax/optional` to create `github.com/openapitools/optional`

- we need to keep `github.com/antihax/optional` as many Go SDKs are still referencing it.
- what about renaming "optional" to "gooptional"? (to make it clearer it's for Go)

Any other thoughts?

cc @OpenAPITools/generator-core-team

**Comment by antihax** [comment]
I am fine with renaming and keeping it around (i am using it anyway).

A shorter name would be nice: goopts? The name will get used a bit in the end developer code.

**Comment by olegsu** [comment]
What is the status of it? 
I just tried to run the golang generator with error: `github.com/antihax/optional@v0.0.0-20180406194304-ca021399b1a6: invalid pseudo-version: does not match version-control timestamp (2018-04-07T02:43:04Z)`

**Comment by wing328** [comment]
Are you using the latest stable version?


在 2019年12月16日週一 下午10:36，Oleg Sucharevich <notifications@github.com> 寫道：

**Comment by olegsu** [comment]
@wing328 latest I guess from example with Docker https://github.com/OpenAPITools/openapi-generator#16---docker
```
docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli generate \
    -i https://g.codefresh.io/api/openapi.json \
    -g go \
    -o /local \
    --additional-properties=withGoCodegenComment=true
```

**Comment by wing328** [comment]
Please do a docker pull first

在 2019年12月18日週三 下午3:42，Oleg Sucharevich <notifications@github.com> 寫道：

**Comment by u23a** [comment]
It's almost 11 months. Is there any updates?

**Comment by wing328** [comment]
The `go` client generator in the latest master has been refactored. Can you give it a try?

**Comment by wing328** [comment]
Closed via #7692
